sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -470,23 +699,370 @@ func getNotifications(d *schema.ResourceData) []MonitorNotification {
 }
 
 func getTriggers(d *schema.ResourceData) []TriggerCondition {
-	rawTriggers := d.Get("triggers").([]interface{})
-	triggers := make([]TriggerCondition, len(rawTriggers))
-	for i := range rawTriggers {
-		triggerDict := rawTriggers[i].(map[string]interface{})
-		triggers[i] = TriggerCondition{
-			TriggerType:     triggerDict["trigger_type"].(string),
-			Threshold:       triggerDict["threshold"].(float64),
-			ThresholdType:   triggerDict["threshold_type"].(string),
-			TimeRange:       triggerDict["time_range"].(string),
-			OccurrenceType:  triggerDict["occurrence_type"].(string),
-			TriggerSource:   triggerDict["trigger_source"].(string),
-			DetectionMethod: triggerDict["detection_method"].(string),
+	if triggerCondition, ok := singletonFromResourceData(d, "trigger_conditions"); ok {
+		ret := triggerConditionsBlockToJson(triggerCondition)
+		return ret
+	} else {
+		rawTriggers := d.Get("triggers").([]interface{})
+		triggers := make([]TriggerCondition, len(rawTriggers))
+		for i := range rawTriggers {
+			triggerDict := rawTriggers[i].(map[string]interface{})
+			triggers[i] = TriggerCondition{
+				TriggerType:     triggerDict["trigger_type"].(string),
+				Threshold:       triggerDict["threshold"].(float64),
+				ThresholdType:   triggerDict["threshold_type"].(string),
+				TimeRange:       triggerDict["time_range"].(string),
+				OccurrenceType:  triggerDict["occurrence_type"].(string),
+				TriggerSource:   triggerDict["trigger_source"].(string),
+				DetectionMethod: triggerDict["detection_method"].(string),
+			}
+		}
+		return triggers
+	}
+}
+
+// 'trigger_conditions' block to TriggerCondition JSON structure
+func triggerConditionsBlockToJson(block map[string]interface{}) []TriggerCondition {
+	conditions := make([]TriggerCondition, 0)
+	if sc, ok := fromSingletonArray(block, logsStaticConditionFieldName); ok {
```

Comment:
nit: do we need error handling for the `ok` values here? (I think the `ok` is error code in this context right?)

Reply:
Maybe, but not super necessary, imo.

The `ok` variable (which is of type `bool`, not an `error`) could be `false` for a field for valid reasons, e.g., when the block does not contain that field. 

That said, a bug in the provider logic could also lead to `ok` being `false` when this function is called on a state, e.g., when the block contains the field, but its value is not in the expected form. Maybe in that case it'll be better to have `fromSingletonArray` panic instead of silently returning `..., false`? 🤔

Reply:
Ended up making `fromSingletonArray` more robust against bad data.